feedback-module/src/lib/hooks/googleAnalytics.js

```diff
@@ -1,15 +1,15 @@
 import { useEffect, useState } from "react";
 import { useGA4React } from "ga-4-react";
 
+import { FORM_EVENTS } from "../constants";
+
 export default function googleAnalytics() {
   const ga = useGA4React();
 
-  const trackFutureResearch = () => {
-    ga &&
-      ga.event(
-        "future_research",
-        "user agreed to participate in research in the future"
-      );
+  const trackFormAction = (formID) => {
```

Comment:
The purpose of this event was solely to track if a user signs up to be a research participant in the future rather than tracking data sent in general

Reply:
I know, I just wanted to avoid hardcoding anything like `if (formID === "research") {...}` and make it more general in case there's any instance in the future where someone might want to track another event related to a form submission— this way we can avoid someone needing to modify the Module.js file.

Reply:
Yes except Mayo requested an event solely for tracking user research because he wanted to know the number just for that form separately from everything else

Reply:
We could have "research" be a variable in the constants file if you think that's better but I think we should still keep a separate event that only tracks user research

Reply:
So right now in constants, I have an object called `FORM_EVENTS` that looks like this:
```
export const FORM_EVENTS = {
  userResearch: {
    event_name: "future_research",
    event: "user agreed to participate in research in the future",
  },
};
```
Right now, this function will still only fire off an event for `future_research`; it's just that in the future if the system admin wanted to add in more GA events related to a form submission, instead of doing something like this:
```
if (formID === "userResearch") {
    trackFutureResearch();
} else if (formID === "someFutureForm") {
    trackSomeFutureEvent();
}...
```
they wouldn't have to touch the code at all and could just edit the constants file.

I might just be over-engineering this a little right now, but I feel like it doesn't hurt to reduce the hard-coding and make things more flexible for a future system admin.

Reply:
Hmm ... I agree we should definitely reduce the hard-coding I'm just wondering if this will make the event less useful inside of the google analytics tracking. Now it won't clearly be an event about future_research, but rather the admin will have to go into every one of the trackFormAction events and look at the specific details just to get the number.

Reply:
Hmmm wdym? Wouldn't this trigger the google analytics tracking in the exact same way & produce events the same way as before? It will still fire an event about future_research.

Reply:
Yeah sorry I misread it, I didn't see that you were tracking the event_name as variable I thought that was hard coded